cmd/migrate.go

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/xataio/pgroll/pkg/migrations"
+)
+
+func migrateCmd() *cobra.Command {
+	var complete bool
+
+	migrateCmd := &cobra.Command{
+		Use:     "migrate <directory>",
+		Short:   "Apply outstanding migrations from a directory to a database",
+		Example: "migrate ./migrations",
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			migrationsDir := args[0]
+
+			m, err := NewRoll(ctx)
+			if err != nil {
+				return err
+			}
+			defer m.Close()
+
+			latestVersion, err := m.State().LatestVersion(ctx, m.Schema())
```

Comment:
can we move all this logic into the pgroll library? Ideally we can use this as a lib too

Reply:
What would the API look like to apply outstanding migrations and where would it live?

something like this?:

```go
m : = roll.New(ctx, pgURL, schema, state, opts)

// Apply outstanding migrations 
err := m.Migrate("./migrations/")
```

or would it be better to have a lower level API that returns unapplied migration files:

```go
m : = roll.New(ctx, pgURL, schema, state, opts)

// get outstanding migration files
err, files := m.UnappliedMigrations("./migrations")
for _, file := range files {
  // start and complete each file
}
```

Do you know how you would like to use this in your existing integration?

Reply:
example on the top would work!

Ideally we work on top of the some `io/fs` abstraction, so I can leverage `embed.FS` as a parameter.